.github/workflows/automate-review-labels.yml

```diff
@@ -35,80 +37,112 @@
                   if (result) {
-                      label = team.replace("@conda-forge/help-", "");
-                      label_no_staged = label.replace("@conda-forge/", "");
+                      const slug = team.replace("@conda-forge/", "");
+                      const label = slug.replace("help-", "");
+                      found_label = true;
                       github.rest.issues.addLabels({
                           issue_number: context.issue.number,
                           owner: context.repo.owner,
                           repo: context.repo.repo,
-                          labels: [label_no_staged, 'review-requested']
-                      })
-                      github.rest.issues.removeLabel({
-                          issue_number: context.issue.number,
-                          owner: context.repo.owner,
-                          repo: context.repo.repo,
-                          name: ['Awaiting author contribution']
+                          labels: [label, 'review-requested']
+                      });
+                      // NOTE: Team review request is required to be posted for
+                      // membership check to work.
+                      github.rest.pulls.requestReviewers({
+                        owner: context.repo.owner,
+                        repo: context.repo.repo,
+                        pull_number: context.issue.number,
+                        team_reviewers: [slug],
                       });
-                      // TODO: Maybe use GitHub API to request review / assign team
                   }
-              }
+              };
+              return found_label
+        - name: remove-labels
+          if: >
+            (steps.check_teams.outputs.result == 'true')
+            && contains(github.event.issue.labels.*.name, 'Awaiting author contribution')
+          uses: actions/github-script@v6
+          with:
+            script: |
+              github.rest.issues.removeLabel({
+                  issue_number: context.issue.number,
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  name: ['Awaiting author contribution']
+              })
+
 
     remove-review-requested-label:
       name: 'When reviewed, remove review-requested label'
       if: >
-        (
-          github.event.issue
-          && github.event.issue.pull_request
-          && (github.event.sender.login != github.event.issue.user.login)
-          && contains(github.event.issue.labels.*.name, 'review-requested')
-        ) || (
-          github.event.pull_request
-          && (github.event.sender.login != github.event.pull_request.user.login)
-          && contains(github.event.pull_request.labels.*.name, 'review-requested')
-        )
+        github.event.pull_request
+        && (github.event.sender.login != github.event.pull_request.user.login)
+        && contains(github.event.pull_request.labels.*.name, 'review-requested')
```

Comment:
We only activate this task when it was triggered by a review because we need the review request to be dismissed.